imagekit-sdk/src/test/java/io/imagekit/sdk/ImageKitTest.java

```diff
@@ -2,629 +2,1890 @@
+		server.enqueue(new MockResponse().setBody("{\n" + "    \"jobId\": \"629f44ac7eb0fe8173622d4b\"\n" + "}"));
+		server.start();
+		RestClient.API_BASE_URL = server.url("/").toString();
+		SUT.moveFolder(moveFolderRequest);
+		RecordedRequest request = server.takeRequest();
+
+		String moveFolderRequestJson = "{\"sourceFolderPath\":\"/testFolder\",\"destinationPath\":\"/Gallery\"}";
+		String utf8RequestBody = request.getBody().readUtf8();
+		assertEquals(moveFolderRequestJson, utf8RequestBody);
+		assertEquals("application/json; charset=utf-8", request.getHeader("Content-Type"));
+		assertEquals("POST /v1/bulkJobs/moveFolder HTTP/1.1", request.getRequestLine());
+		assertEquals(RestClient.API_BASE_URL.concat("v1/bulkJobs/moveFolder"), request.getRequestUrl().toString());
+	}
+
+	@Test
+	public void getBulkJobStatus_500_Expected()
+			throws InterruptedException, IOException, ForbiddenException, TooManyRequestsException,
+			InternalServerException, UnauthorizedException, BadRequestException, UnknownException {
+
+		MockWebServer server = new MockWebServer();
+		server.enqueue(new MockResponse().setResponseCode(500));
+		server.enqueue(new MockResponse().setBody(
+				"{\n" + "    \"message\": \"We have experienced an internal error while processing your request.\",\n"
+						+ "    \"help\": \"For support kindly contact us at support@imagekit.io .\"\n" + "}"));
+		server.start();
+		RestClient.API_BASE_URL = server.url("/").toString();
+		ResultBulkJobStatus resultBulkJobStatus = SUT.getBulkJobStatus("jobId");
+		RecordedRequest request = server.takeRequest();
+
+		assertEquals(500, resultBulkJobStatus.getResponseMetaData().getHttpStatusCode());
+		assertEquals("application/json", request.getHeader("Content-Type"));
+		assertEquals("GET /v1/bulkJobs/jobId HTTP/1.1", request.getRequestLine());
+		assertEquals(RestClient.API_BASE_URL.concat("v1/bulkJobs/jobId"), request.getRequestUrl().toString());
+	}
+
+	@Test
+	public void getBulkJobStatus_successExpected()
+			throws InterruptedException, IOException, ForbiddenException, TooManyRequestsException,
+			InternalServerException, UnauthorizedException, BadRequestException, UnknownException {
+
+		MockWebServer server = new MockWebServer();
+		server.enqueue(new MockResponse().setBody("{\n" + "    \"jobId\": \"629f44ac7eb0fe8173622d4b\",\n"
+				+ "    \"type\": \"MOVE_FOLDER\",\n" + "    \"status\": \"Completed\"\n" + "}"));
+		server.start();
+		RestClient.API_BASE_URL = server.url("/").toString();
+		SUT.getBulkJobStatus("629f44ac7eb0fe8173622d4b");
+		RecordedRequest request = server.takeRequest();
+
+		assertEquals("application/json", request.getHeader("Content-Type"));
+		assertEquals("GET /v1/bulkJobs/629f44ac7eb0fe8173622d4b HTTP/1.1", request.getRequestLine());
+		assertEquals(RestClient.API_BASE_URL.concat("v1/bulkJobs/629f44ac7eb0fe8173622d4b"),
+				request.getRequestUrl().toString());
+	}
+
+	@Test(expected = NotFoundException.class)
+	public void getFileVersions_404_Expected()
```

Comment:
Missing test case for `updateDetail`

Reply:
They are already present. Check these functions: imageKit_updateDetails_*